pkg/hoist/hoist_launchable.go

```diff
@@ -24,19 +24,20 @@ import (
 type Launchable struct {
-	Location         string              // A URL where we can download the artifact from.
-	Id               string              // A (pod-wise) unique identifier for this launchable, used to distinguish it from other launchables in the pod
-	ServiceId        string              // A (host-wise) unique identifier for this launchable, used when creating runit services
-	RunAs            string              // The user to assume when launching the executable
-	PodEnvDir        string              // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
-	Fetcher          uri.Fetcher         // Callback that downloads the file from the remote location.
-	RootDir          string              // The root directory of the launchable, containing N:N>=1 installs.
-	P2Exec           string              // Struct that can be used to build a p2-exec invocation with appropriate flags
-	ExecNoLimit      bool                // If set, execute with the -n (--no-limit) argument to p2-exec
-	CgroupConfig     cgroups.Config      // Cgroup parameters to use with p2-exec
-	CgroupConfigName string              // The string in PLATFORM_CONFIG to pass to p2-exec
-	RestartTimeout   time.Duration       // How long to wait when restarting the services in this launchable.
-	RestartPolicy    runit.RestartPolicy // Dictates whether the launchable should be automatically restarted upon exit.
+	Location         string                 // A URL where we can download the artifact from.
+	Id               string                 // A (pod-wise) unique identifier for this launchable, used to distinguish it from other launchables in the pod
+	ServiceId        string                 // A (host-wise) unique identifier for this launchable, used when creating runit services
+	RunAs            string                 // The user to assume when launching the executable
+	PodEnvDir        string                 // The value for chpst -e. See http://smarden.org/runit/chpst.8.html
+	Fetcher          uri.Fetcher            // Callback that downloads the file from the remote location.
+	RootDir          string                 // The root directory of the launchable, containing N:N>=1 installs.
+	P2Exec           string                 // Struct that can be used to build a p2-exec invocation with appropriate flags
+	ExecNoLimit      bool                   // If set, execute with the -n (--no-limit) argument to p2-exec
+	CgroupConfig     cgroups.Config         // Cgroup parameters to use with p2-exec
+	CgroupConfigName string                 // The string in PLATFORM_CONFIG to pass to p2-exec
+	RestartTimeout   time.Duration          // How long to wait when restarting the services in this launchable.
+	RestartPolicy    runit.RestartPolicy    // Dictates whether the launchable should be automatically restarted upon exit.
+	SuppliedEnvVars  map[string]interface{} // A map of user-supplied environment variables to be expored for this launchable
```

Comment:
Environment variables are definitely strings, so let's have `map[string]string`

---

I agree that these are definitely "user-supplied" variables, but why not keep the name simple: `EnvVars`?